src/rust/src/backend/kdf.rs

```diff
@@ -401,35 +412,47 @@ impl Argon2id {
     fn derive_phc_encoded<'p>(
         &mut self,
         py: pyo3::Python<'p>,
+        variant: &Argon2Variant,
         key_material: CffiBuf<'_>,
     ) -> CryptographyResult<pyo3::Bound<'p, pyo3::types::PyString>> {
-        let derived_key = self.derive(py, key_material)?;
+        let derived_key = self.derive(py, variant, key_material)?;
         let salt_bytes = self.salt.as_bytes(py);
 
         let salt_b64 = STANDARD_NO_PAD.encode(salt_bytes);
         let hash_b64 = STANDARD_NO_PAD.encode(derived_key.as_bytes());
 
+        let variant_id: &str = match variant {
+            Argon2Variant::Argon2d => "argon2d",
+            Argon2Variant::Argon2i => "argon2i",
+            Argon2Variant::Argon2id => "argon2id",
+        };
+
         // Format the PHC string
         let phc_string = format!(
-            "$argon2id$v=19$m={},t={},p={}${}${}",
-            self.memory_cost, self.iterations, self.lanes, salt_b64, hash_b64
+            "${}$v=19$m={},t={},p={}${}${}",
+            variant_id, self.memory_cost, self.iterations, self.lanes, salt_b64, hash_b64
         );
 
         Ok(pyo3::types::PyString::new(py, &phc_string))
     }
 
     #[cfg(CRYPTOGRAPHY_OPENSSL_320_OR_GREATER)]
-    #[staticmethod]
-    #[pyo3(signature = (key_material, phc_encoded, secret=None))]
     fn verify_phc_encoded(
         py: pyo3::Python<'_>,
+        variant: &Argon2Variant,
         key_material: CffiBuf<'_>,
         phc_encoded: &str,
         secret: Option<pyo3::Py<pyo3::types::PyBytes>>,
     ) -> CryptographyResult<()> {
         let parts: Vec<_> = phc_encoded.split('$').collect();
 
-        if parts.len() != 6 || !parts[0].is_empty() || parts[1] != "argon2id" {
+        let variant_id: &str = match variant {
+            Argon2Variant::Argon2d => "argon2d",
+            Argon2Variant::Argon2i => "argon2i",
+            Argon2Variant::Argon2id => "argon2id",
+        };
+
+        if parts.len() != 6 || !parts[0].is_empty() || parts[1] != variant_id {
             return Err(CryptographyError::from(exceptions::InvalidKey::new_err(
                 "Invalid PHC string format.",
```

Comment:
If this is is an invalid variant (e.g., they used the wrong class), should we emit a more instructive error?